### PR TITLE
Fix `pub get --unknown-flag`

### PR DESCRIPTION
**Pull request by sigurdm**
Fixes argument handling in `flutter pub get` when a single flag is passed.

https://github.com/flutter/flutter/issues/119614

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/hermetic/pub_get_test.dart
Comment (jonasfj) [inline packages/flutter_tools/test/commands.shard/hermetic/pub_get_test.dart:210]: wow, this is not pretty :see_no_evil: 

But I guess it's a way to test things.

**Review by jonasfj — APPROVED** [review]
LGTM, but then again, I don't know much about `flutter_tools`, so I'll defer to @christopherfujino :D

File: packages/flutter_tools/lib/src/commands/packages.dart
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/packages.dart:216]: Generally in the tool, we handle the `verbose` flag separately, before the arg parser: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/executable.dart#L64

Do we need to handle this here?

**Review by christopherfujino — APPROVED** [review]
one question about if we need to parse `-v`, otherwise LGTM

File: packages/flutter_tools/lib/src/commands/packages.dart
Comment (sigurdm) [reply packages/flutter_tools/lib/src/commands/packages.dart:216]: It is not really being handled - just accepted by this preliminary parser.
Otherwise we would end up not understanding eg. '--dry-run -v'.

**Pull request merged**
